### ISSUE TITLE
[Feature] Manage Applications from CI/CD

### ISSUE THREAD

**Issue by shan916** [Axway-API-Management-Plus/apimanager-swagger-promote#180]
**User story**
> To integrate Swagger-Promote into our CI/CD pipeline we would like to have
the ability to manage Applications.

**Additional context**
> It would be useful to be able to manage applications from a CI/CD pipeline. The use case is when a developer wants to on-board a brand new API to API Manager along with a brand new application. This currently requires that the application be added manually in API Manager before swagger-promote can be used.

**Comment by gcornacchia** [comment]
we are also interested in this feature (also for managing organizations), @cwiechmann have you already thought how to approach this topic? if you want we can help writing some code...

**Comment by cwiechmann** [comment]
Hi @gcornacchia, Yes, I'm thinking to make this possible using a possible Plugin-Approach. So, for that, a Plugin would take over and for instance right at the beginning creating an application, before the API is then replicated. See here what I have mind to support with plugins:
https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/blob/plugins/plugins/README.md
However, besides that specification I haven't started working on it. I'm happy to get your feedback.

**Comment by cwiechmann** [comment]
Created a dedicated feature-request #190 for Plugin-Support, which can be used to add Application-Handling

**Comment by gcornacchia** [comment]
HI, I took a look at your links and plugins branch code. From what I understood the plugin is all about the api lifecycle (after the api is loaded, after the desired api has loaded, before the changes replication and post process). I imagined a workflow like this:
* 1) I align API Manager with all organizations and applications
* 2) I align API Manager with all APIs (a lot of APIs) with grants and permissions
* 3) I use the APIs

In this scenario I would like to have all the data definining organizations and applications in separated json files from API config files.
So the plugin appoach is a nice and usefull feature but I imagine more extension points than the 4 already defined. Moreover a plugin could define additional command line parameters and do somebatch tasks (like align a kps table with data, change api manager settings,...) without touching the virtualized apis... 

just my thaughts...
